components/button/button-icon.js

```diff
@@ -100,6 +101,12 @@ class ButtonIcon extends ThemeMixin(ButtonMixin(VisibleOnAncestorMixin(RtlMixin(
 				:host([active]) button {
 					background-color: var(--d2l-button-icon-background-color-hover);
 				}
+				:host d2l-icon {
+					color: var(--d2l-button-icon-text-color);
+				}
+				:host button:hover > d2l-icon {
```

Comment:
Should this be applied in the focus state as well?

Reply:
Nice catch! It does look like we are applying hover styling to the focus state for button-icons.